app/javascript/serviceworker/messages.js

```diff
@@ -10,15 +10,24 @@ const messageHandlers = {
     event.ports[0].postMessage(isOnline());
   },
 
-  SAVE_CAMPAIGN_FOR_OFFLINE: async ({ data }) => {
-    const campaignId = data.payload["campaignId"];
+  SAVE_CAMPAIGN_FOR_OFFLINE: async (event) => {
+    const campaignId = event.data.payload["campaignId"];
+    const password = event.data.payload["password"];
 
-    addAll([
+    await initCache(password);
+
+    await addAll([
+      ...event.data.payload["additionalItems"],
+      `/favicon.ico`,
+      `/dashboard`,
```

Comment:
This part concerns me a little ... I think the access codes are per-campaign. It's possible a SAIS nurse will be working on more than one offline campaign at a time, esp if they're part of a big team. I think for security purposes, seeing as each campaign is a different data set we should have a different access code for each campaign.

That being the case, I don't think assets that are used across assets should be encrypted. For example, which password would they enter to view /dashboard?

In any case, probably for another PR, I'll create an issue.